client/src/pages/Meals.js

```diff
@@ -1,35 +1,94 @@
 // TEMPORARY PAGE WITH RECIPES WILL BE REMOVED WHEN CHEF PROFILE PAGE IS CREATED
-import React from "react";
+import React, { useState, useEffect } from "react";
+import makeStyles from "@material-ui/core/styles/makeStyles";
 import ResponsiveSideBar from "../components/ResponsiveSideBar";
 import Main from "../components/Main";
-import { Typography } from "@material-ui/core";
+import { Typography, Grid, Button } from "@material-ui/core";
 import MealCard from "../components/MealCard";
-import meals from "../lib/mockedMeals";
+import useCuisineSelector from "../lib/useCuisineSelector";
+import { abortableFetch, consoleErrorNonAbortErrors } from "../utils";
+
+const useStyles = makeStyles({
+    availableMeals: { paddingBottom: "15px", display: "inline", fontSize: "30px" },
+});
 
 function Meals(props) {
+    const classes = useStyles();
+    const {
+        ALL,
+        cuisines,
+        availableCuisines,
+        addCuisine,
+        removeCuisine,
+    } = useCuisineSelector();
+
+    const [meals, setMeals] = useState([]);
+
+    useEffect(() => {
+        let url = "/search?searchType=meals";
+        if (!cuisines.has(ALL)) {
+            url = `${url}&cuisine=` + [...cuisines].join(",");
+        }
+        const { ready, abort } = abortableFetch(url);
+        ready
+            .then((res) => res.json())
+            .then((res) => res.meals)
+            .then(setMeals)
+            .catch(consoleErrorNonAbortErrors);
+        return abort;
+    }, [cuisines]);
+
     return (
-        <React.Fragment>
+        <Main>
             <ResponsiveSideBar>
-                <Typography>Responsive Side Bar</Typography>
+                <Grid container spacing={2} alignContent="flex-start">
+                    <Grid item xs={12}>
+                        <Typography>Cuisine:</Typography>
+                        {[...cuisines].map((cuisine) => (
```

Comment:
Do we need `[...cuisines]` everywhere instead of just `cuisines`?

Reply:
Yes, we need it to convert the set into a list